src/otx/core/model/entity/base.py

```diff
@@ -233,10 +234,9 @@ def __init__(self, num_classes: int, config: DictConfig) -> None:
         self.config = config
         super().__init__(num_classes)
 
-    def _create_model(self) -> nn.Module:
+    def _create_model(self, configuration: dict[str, Any] | None = None) -> Model:
```

Comment:
1. Please do not use `None`able typing as possible if there is no special issue. In this case, you can use `configuration: dict[str, Any] = {}`
2. However, please do not change the function signature of `_create_model(self)` derived from the base model if there is no special reason. In this case, I can see L253
```python
return Model.create_model(model_adapter, model_type=self.model_type, configuration=configuration)
```
One configuration is passed from the member variable (`self.model_type`), but another comes from the function argument `configuration`. Please keep one single channel for it.

Reply:
1. configuration: dict[str, Any] = {} raise mypy issue
2.  Please, see my new updates. OVModel isn't inherited from OTXModel

Reply:
What's the reason that we should maintain two `OTXModel` and `OVModel` base classes although they share 99% of codes?

Reply:
There are many reasons. They don't share 99% of code, unfortunately, every change now brings more and more gap between these two entitites.
1) Currently, OVModel looks like ugly Workaround to fit OTXModel. After CLI-API changes, it is really hard to work with it in in its current form. We should use config and support ugly structure for that, it is now hard to reload parameters for OVModel via CLI. We need to change constructor and put all of the configuration parameters there to have clear API and easy modification.
2) OVModel isn't nn.Module. It doesn't support schedulers, optimizers, training and validation phase at all. It doesn't need to support class incremental and all methods related to checkpoint loading.
It is completely different entity meanwhile it share "naming" of the methods, but not the nature.

Going further, I would like to divide LitModule as well and make OVModel independent and easy to modify and understand. It is required more changes and I would like to make a proposal on our OTX forum next Tuesday
For now I propose to at least divide OVModel   and modify constructor to make our work with OpenVINO functionality easier.

Reply:
> They don't share 99% of code

I can see that you already copied many same code pieces from `OTXModel` here
https://github.com/openvinotoolkit/training_extensions/blob/a39a7a9baee91e06a6b446e6081a1e1309c51a2d/src/otx/core/model/entity/base.py#L212-L314

> Currently, OVModel looks like ugly Workaround to fit OTXModel. After CLI-API changes, it is really hard to work with it in in its current form. We should use config and support ugly structure for that, it is now hard to reload parameters for OVModel via CLI. We need to change constructor and put all of the configuration parameters there to have clear API and easy modification.

What's the problem on
```python
class OVModel(OTXModel, Generic[T_OTXBatchDataEntity, T_OTXBatchPredEntity]):
    def __init__(
        self,
        num_classes: int,
        model_name: str,
        ...):
        super().__init__(num_classes=num_classes)
        ...
```
? You can change the constructor of `OVModel` freely.

> OVModel isn't nn.Module. It doesn't support schedulers, optimizers, training and validation phase at all. It doesn't need to support class incremental and all methods related to checkpoint loading.
It is completely different entity meanwhile it share "naming" of the methods, but not the nature.

The reason why `OVModel` should be derived from `OTXModel` is that it can shares data pipelines, evaluation loop, and metric computations in a single system. I still don't know what is the exact disadvantage of inheriting from `nn.Module` because the computational cost of deriving from `nn.Module` is ignorable. But, if you really doesn't like `nn.Module`, we might consider changing `OTXModel` as a pure Python class and introducing an additional layer such as

```python
class OTXModel:
     ...

class OTXTorchModel(OTXModel, nn.Module):
     ...

class OVModel(OTXModel):
     ...
```

> Going further, I would like to divide LitModule as well and make OVModel independent and easy to modify and understand.

I'm really against on maintaining two different systems in one product. We already had many lessons from OTX 1.x.

Reply:
I agree with you considering maintenance of two systems. However, I want to make OVModel more independent and clear entity without any export, train functionality. 
Maybe, creating base meta class is a good option. I will review possibility to make OVModel more independent, but without creatin new LitModule

For now, let's change constructor for OVModel. It is essential to avoid ugly initialization of this class and change easily via CLI arguments . I made corresponding changes.